DOC - Enhance installation, running and visualizing benchmarks (#629)
benchopt/benchopt: doc/get_started.rst

Get started
===========

Installation
------------

> The recommended way to use benchopt is within a conda environment to fully-benefit from all its features.

Reviewer: fully benefit without hyphen?

Hence, start by creating a dedicated conda environment and then activate it.

.. prompt:: bash $

    conda create -n benchopt python
    conda activate benchopt

Benchopt is available on PyPi. You can get the **stable version** via ``pip`` by running the following command

.. prompt:: bash $

    pip install -U benchopt

> Eager to try out the **development version**? you can run instead

Reviewer: missing upper case

.. prompt:: bash $

    pip install -U -i https://test.pypi.org/simple/benchopt

.. attention::

   The **development version** is a work in progress and hence might contain incomplete features.
   A typical user is advised to use the **stable version** instead.

With benchopt being installed, you get access to the :ref:`Command Line Interface (CLI) <cli_ref>`,

> which enables simple and easy manipulation of benchmarks just from the terminal.

Reviewer: ```suggestion
which enables running benchmarks from the terminal.
```

Author: The CLI is not only for running benchmarks, it is also for installation, publishing ...

I would keep
```suggestion
which enables a simple manipulation of benchmarks from the terminal.
```

Author: I'm -1 with removing this part. It is important to hint that the proper way to use `benchopt` is through the CLI.

As a side note, getting feedback from one user is to be considered not to be directly implemented. It is tricky to please everyone.

Reviewer: yes, that's why I asked to remove the "running a benchmark through a python file" :) all the doc uses the benchopt cli tool
readding a removed sentence on the benchopt CLI :)

Run an existing benchmark
-------------------------

Let's get the first steps with benchopt by comparing some solvers of

> `Lasso problem <https://en.wikipedia.org/wiki/Lasso_(statistics)>`_ on a

Reviewer: a/the?

`Leukemia dataset <https://www.science.org/doi/10.1126/science.286.5439.531>`_.

> Benchopt community maintains :ref:`several optimization benchmarks <available_benchmarks>`

Reviewer: IMO this information can be ommitted here, documentation should go straight to the point.

Author: I think it prepares well the floor for using Benchmark Lasso?
Also, It points to the ecosystem of benchopt benchmark, which IMO valuable for the visibility of this benchmarks reops.

Reviewer: yesterday's experience with Nelly hinted that there was too much information on the page, which make critical information harder to find. I'll remove this part
The other benchmarks are exposed on the repo's readme

> and thrives at making them accessible and up to date, so as for the Lasso problem.

Reviewer: thrive to + V / thrive at + N  IIRC

Start by cloning the Lasso benchmark repository and then ``cd`` to it.

.. prompt:: bash $

    git clone https://github.com/benchopt/benchmark_lasso.git
    cd benchmark_lasso

Then install automatically the benchmark requirements, namely, the solvers `skglm <https://contrib.scikit-learn.org/skglm/>`_ and

> `scikit-learn <https://scikit-learn.org/stable/>`_, and then the dataset Leukemia. 

Reviewer: why not do it in one call?

Author: Yess, indeed! That would be much simplier

Get started
===========

Installation
------------

The recommended way to use benchopt is within a conda environment to fully-benefit from all its features.
Hence, start by creating a dedicated conda environment. 

.. code-block:: bash

    # create conda environment with name benchopt
    conda create -n benchopt python

    # activate the environment
    conda activate benchopt

Benchopt is available on PyPi. You can get the **stable version** via ``pip`` by running the following command

> .. code-block:: bash

Reviewer: Using `. prompt:: bash $` as before puts a dollar at the beginning of each line, making it clearer that it's a block of bash code. Can you revert?

Reviewer: Also there's a nice feature where on the top right corner of the box there's a button that you can click to copy the snippet, if you manage to implement it :pray: https://sphinx-copybutton.readthedocs.io/en/latest/
